components/compliance-service/reporting/util/csv_test.go

```diff
@@ -14,6 +16,16 @@ const (
+	returnedString := maxCharLimit(my5kstring)
+	assert.Equal(t, my5kstring, returnedString)
+
+	returnedString = maxCharLimit(my5kstring + my5kstring + my5kstring + my5kstring + my5kstring + my5kstring + my5kstring)
+	assert.Equal(t, strings.HasPrefix(returnedString, "character limit exceeds 32,000. truncating message"), true)
+	assert.Equal(t, utf8.RuneCountInString(returnedString) < 32000, true)
+}
```

Comment:
I think it'd be valuable to test the limits directly, by having one test that is exactly the amount allowed and I've that is exactly one character too many.

The strings could also be generated in a loop, that might be easier to work with than just a huge string literal.

Reply:
ya, that makes sense!